### PR TITLE
utils: add janus_strlcat helper

### PR DESCRIPTION
**Pull request by tmatth**
This will log an error if the output is truncated.

### CODE REVIEW

File: utils.c [hunk 1]
```diff
@@ -1217,3 +1217,14 @@ size_t janus_gzip_compress(int compression, char *text, size_t tlen, char *compr
 	return zs.total_out;
 }
 #endif
+
+
+gsize janus_strlcat(gchar *dest, const gchar *src, gsize dest_size)
+{
```
Comment (lminiero) [inline utils.c:1223]: Just a code style nit: please put opening bracket on same line as function.

File: postprocessing/janus-pp-rec.c [hunk 2]
```diff
@@ -117,6 +117,7 @@ Usage: janus-pp-rec [OPTIONS] source.mjr
 #include <jansson.h>
 
 #include "../debug.h"
+#include "../utils.h"
```
Comment (lminiero) [inline postprocessing/janus-pp-rec.c:120]: I'll have to check if this means we need to add `utils.o` as an object when building `janus-pp-rec`. I see the automated Github Actions checks have all succeeded, so this shouldn't be an issue.

**Review by lminiero — COMMENTED** [review]
Apart from a tiny code-style nit (I know, I'm the worst :laughing: ), this looks like a good fence to me. I already know it will cause me more than a little headache when re-aligning multistream (it will likely cause many conflicts), but that's indeed needed. I'll make a couple of tests, then merge and address the multistream conflicts so that we can start addressing the SDP size there.

File: postprocessing/janus-pp-rec.c (hunk 2 above)
Comment (tmatth) [reply postprocessing/janus-pp-rec.c:120]: `utils.c` is already listead as part of `janus_pp_rec_SOURCES` in `Makefile.am` so I think there's no change really.

**Comment by lminiero** [comment]
Thanks for the quick fix! Merging then, and heading down the multistream merge conflicts hole...

**Pull request merged**